### PR TITLE
Add support for running a single Flutter test, by name

### PR DESCRIPTION
**Pull request by skybrian**
Depends on:
https://github.com/flutter/flutter/pull/11020

### CODE REVIEW

**Comment by skybrian** [comment]
This shouldn't be landed until the Flutter change lands, but I think it's getting close.

File: src/io/flutter/run/test/TestFields.java [hunk 1]
```diff
@@ -28,41 +28,68 @@
...
   /**
    * Returns a value indicating whether we're running tests in a file or in a directory.
    */
   @NotNull
   public Scope getScope() {
-    return testFile != null ? Scope.FILE : Scope.DIRECTORY;
+    if (testName != null) {
+      return Scope.NAME;
+    } else if (testFile != null) {
+        return Scope.FILE;
```
Comment (devoncarew) [inline src/io/flutter/run/test/TestFields.java]: nit: formatting

**Comment by devoncarew** [comment]
lgtm (once the associated flutter PR lands)

File: src/io/flutter/run/test/TestFields.java (hunk 1 above)
Comment (skybrian) [inline src/io/flutter/run/test/TestFields.java]: Will fix in the next CL.

**Pull request merged**